RestaurantReservation.Db/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.DataModels;
+using RestaurantReservation.Db.IServices;
+using RestaurantReservation.Db.Utilities;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class OrderRepository : IRepositoryServices<Order, string>, IOrderServices
+    {
+        private readonly RestaurantReservationDbContext _dbContext;
+
+        public OrderRepository(RestaurantReservationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<OperationResult<string>> AddAsync(Order order)
+        {
+            try
+            {
+                await _dbContext.Orders.AddAsync(order);
+                await _dbContext.SaveChangesAsync();
+                return OperationResult<string>.SuccessResult($"id:{order.Id}");
+            }
+            catch (DbUpdateException ex) when (IsForeignKeyViolation(ex))
+            {
+                return OperationResult<string>.FailureResult("Foreign key constraint violation. Make sure associated entities exist.");
+            }
+            catch (Exception ex)
+            {
+                return OperationResult<string>.FailureResult($"An error occurred while processing the order.{ex.Message}");
+            }
+
+            bool IsForeignKeyViolation(DbUpdateException ex)
```

Comment:
the second time I see it, you could move it as  extension method for the DbUpdateException